src/FakeItEasy/Core/DefaultArgumentConstraintManager.cs

```diff
@@ -3,79 +3,66 @@ namespace FakeItEasy.Core
     using System;
 
     internal class DefaultArgumentConstraintManager<T>
-        : INegatableArgumentConstraintManager<T>
+        : ICapturableArgumentConstraintManager<T>
     {
-        private readonly Action<IArgumentConstraint> onConstraintCreated;
+        private readonly Action<MatchesConstraint> onConstraintCreated;
 
         public DefaultArgumentConstraintManager(Action<IArgumentConstraint> onConstraintCreated)
+            : this((MatchesConstraint constraint) => onConstraintCreated(constraint))
         {
-            this.onConstraintCreated = onConstraintCreated;
         }
 
-        public IArgumentConstraintManager<T> Not => new NotArgumentConstraintManager(this);
+        private DefaultArgumentConstraintManager(Action<MatchesConstraint> onConstraintCreated) =>

```

Comment:
I was a bit confused by this...
Do we need to keep the old constructor? It's only used in `RootModule.ArgumentConstraintManagerFactory` (and in tests). We could just change the signature of `ArgumentConstraintTrap.ReportTrappedConstraint`.

Reply:
Never mind that. Not all `IArgumentConstraint`s are `MatchesConstraint`s...